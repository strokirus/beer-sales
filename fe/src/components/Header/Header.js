import React from 'react';
import { string } from 'prop-types';
import {
  Link,
} from 'react-router-dom';

import {
  MainHeader,
} from './styles';

import Search from '../../containers/Search';

/**
 * Component to Header
 * @param term Term to Search
*/
const Header = ({ term }) => (
  <MainHeader>
    <Link
      to="/"
    >
      <h1>Beer Store</h1>
    </Link>
    <Search
      term={term}
    />
  </MainHeader>
);

Header.propTypes = {
  term: string,
};

Header.defaultProps = {
  term: '',
};

export default Header;
