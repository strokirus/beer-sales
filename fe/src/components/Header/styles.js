import styled from 'styled-components';

import {
  colors,
} from '../../cssGlobals/variables';

import { breakpoints } from '../../helpers/breakpoints';

export const MainHeader = styled.header`
  padding: 0 25px;
  width: 100%;
  box-shadow: rgba(0, 0, 0, 0.15) 0px 2px 4px 0px;
  margin-bottom: 10px;
  line-height: 60px;
  height: 60px;
  display: grid;
  grid-template-columns: 20% 75%;
  grid-template-rows: auto auto;
  background-color:
    ${({ theme }) => (theme.mode === 'dark' ? `${colors.dark}` : `${colors.white}`)};

  a {
    text-decoration: none;
  }
  
  h1 {
    font-size: 20px;
    font-weight: 500;
  }

  @media (max-width: ${breakpoints.small}) {
    grid-template-columns: 40% 60%;
  }
`;

export default {
  MainHeader,
};
