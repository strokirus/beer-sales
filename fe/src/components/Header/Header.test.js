import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Header from './Header';
import store from '../../store';

describe('(Component) Header', () => {
  it('should render an Header', () => {
    const tree = renderer
      .create(
        <Provider store={store}><BrowserRouter><Header /></BrowserRouter></Provider>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
