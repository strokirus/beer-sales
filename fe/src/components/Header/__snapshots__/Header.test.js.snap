// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Header should render an Header 1`] = `
<header
  className="sc-AxjAm fgdJHj"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <h1>
      Beer Store
    </h1>
  </a>
  <input
    className="sc-AxirZ cwgdyS"
    id="search-box"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Search your beer here..."
    type="text"
    value=""
  />
  <label
    className="sc-AxiKw gShybA"
    htmlFor="search-box"
  >
    Search your beers MainHeader
  </label>
</header>
`;
